examples/cloud-composer-examples/composer_http_post_example/ephemeral_dataproc_spark_dag.py

```diff
@@ -132,17 +126,18 @@
     # Delete  gcs files in the timestamped transformed folder.
     delete_transformed_files = BashOperator(
         task_id='delete_transformed_files',
-        bash_command="gsutil -m rm -r gs://" + BUCKET +
+        bash_command="gsutil -m rm -r gs://{{ var.value.gcs_bucket }}" +
         "/{{ dag_run.conf['transformed_path'] }}/")
 
     # If the spark job or BQ Load fails we rename the timestamped raw path to
     # a timestamped failed path.
-    move_failed_files = BashOperator(task_id='move_failed_files',
-                                     bash_command="gsutil mv gs://" + BUCKET +
-                                     "/{{ dag_run.conf['raw_path'] }}/ " +
-                                     "gs://" + BUCKET +
-                                     "/{{ dag_run.conf['failed_path'] }}/",
-                                     trigger_rule=TriggerRule.ONE_FAILED)
+    move_failed_files = BashOperator(
```

Comment:
IMO no.
What is the value in moving the files?

The point of this example is being able to have another process manually/directly trigger a DAG with REST API (e.g. once it is done dumping the raw files to a known prefix). The REST payload contains the prefix. If this other process has a the ability to control the prefix it writes to then there's no reason to re-use a prefix and leaving the files where they are make audit trail,  troubleshooting and certain replay scenarios easier.

Imagine you have a large body of data (transaction history) that you enhance with a look up table (side input that has latest user segmentation). It could be reasonable to say "I provide the currator of the look up table a REST API to re-run this DAG that runs this enhancement on the same data with their new look up values". This use case would be unecissarily complicated if this look up table gets updated rather frequently and we always moved the files to a success dir every time. This isn't really a backfill or rerun of a partition like we normally think of in airflow. To be blunt orchestration like [prefect](https://docs.prefect.io/) might better serve this use case but few people have those skills and GCP doesn't have a managed service for that 😃. 

A more interesting use of this pattern might be a dataflow streaming pipeline that performs some windowed aggregation before writing to target table. It could handle super late events by buffering them to files on GCS and kicking off an airflow DAG that's responsible for updating said aggregations in light of these super late events in the target table. It might be helpful to look directly at the buffered files where they are for troubleshooting and fire the same REST call to trigger the DAG again if necessary.

It's really dependent on use case if you want to move these files (imagine if there are other consumers of these input files that don't expect them to be moved).

Reply:
I agree that it depends on the use case how the files should be treated post processing..